crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -572,6 +572,90 @@ impl Mesh {
         Ok(self)
     }
 
+    /// Merges the [`Mesh`] data of `other` with `self`. The attributes and indices of `other` will be appended to `self`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the vertex attribute values of `other` are incompatible with `self`.
+    /// For example, [`VertexAttributeValues::Float32`] is incompatible with [`VertexAttributeValues::Float32x3`].
+    pub fn merge(&mut self, other: Mesh) {
+        use VertexAttributeValues::*;
+
+        // The indices of `other` should start after the last vertex of `self`.
+        let index_offset = self.count_vertices();
+
+        // Extend attributes of `self` with attributes of `other`.
+        for (id, values) in self.attributes_mut() {
+            let enum_variant_name = values.enum_variant_name();
+            if let Some(other_values) = other.attribute(id) {
+                match (values, other_values) {
```

Comment:
From what I can see, if `self` and `other`'s attributes don't match up:

- if `self` has an attribute but `other` doesn't, it is kept (note that this means future's `self.count_vertices()` will report the old vertices count);
- if `other` has an attribute but `self` doesn't, it is silently ignored.

Which could be kinda surprising. I don't see much better ways to handle this (other than error), in both cases it should be documented.

Reply:
For now, I just added a short mention in the doc comment that attributes of `other` that don't exist on `self` will be ignored